File: app/src/main/java/com/stevenfrew/beatprompter/bluetooth/SignalOnlyBluetoothMessage.java
package com.stevenfrew.beatprompter.bluetooth;

abstract class SignalOnlyBluetoothMessage extends BluetoothMessage
{
    private byte mMessageID;
    SignalOnlyBluetoothMessage(byte messageID)
    {
        mMessageID=messageID;
    }

    SignalOnlyBluetoothMessage(byte[] bytes) throws NotEnoughBluetoothDataException
    {
        if(bytes.length<1)
            throw new NotEnoughBluetoothDataException();
        mMessageLength=1;
    }

    public byte[] getBytes()
    {
        return new byte[]{mMessageID};
    }
}

File: app/src/main/kotlin/com/stevenfrew/beatprompter/bluetooth/message/SignalOnlyBluetoothMessage.kt
package com.stevenfrew.beatprompter.bluetooth.message

/**
 * Base class for Bluetooth messages that are "signal only", i.e. they have no accompanying data.
 */
abstract class SignalOnlyBluetoothMessage constructor(messageID: Byte) : BluetoothMessage() {
    private val mMessageID=messageID

    override val bytes: ByteArray
        get() = byteArrayOf(mMessageID)
}
